cmd/juju/cloud/showcredential.go

```diff
@@ -0,0 +1,181 @@
+)
+
+type showCredentialCommand struct {
+	modelcmd.CommandBase
+	store jujuclient.ClientStore
+
+	out cmd.Output
+
+	newAPIFunc func() (CredentialContentAPI, error)
+
+	CloudName      string
+	CredentialName string
+
+	ShowSecrets bool
+}
+
+// NewShowCredentialCommand returns a command to show information about
+// credentials stored on the controller.
+func NewShowCredentialCommand() cmd.Command {
+	cmd := &showCredentialCommand{
+		store: jujuclient.NewFileClientStore(),
+	}
+	cmd.newAPIFunc = func() (CredentialContentAPI, error) {
+		return cmd.NewCredentialAPI()
+	}
+	return modelcmd.WrapBase(cmd)
+}
+
+func (c *showCredentialCommand) SetFlags(f *gnuflag.FlagSet) {
+	c.CommandBase.SetFlags(f)
+	// We only support yaml for display purposes.
+	c.out.AddFlags(f, "yaml", map[string]cmd.Formatter{
+		"yaml": cmd.FormatYaml,
+	})
+	f.BoolVar(&c.ShowSecrets, "show-secrets", false, "Display credential secret attributes")
+}
+
+func (c *showCredentialCommand) Init(args []string) error {
+	switch len(args) {
+	case 0:
+		// will get all credentials stored on the controller for this user.
+		break
+	case 1:
+		return errors.New("both cloud and credential name are needed")
+	case 2:
+		c.CloudName = args[0]
+		c.CredentialName = args[1]
+	default:
+		return errors.New("only cloud and credential names are supported")
+	}
+	return nil
+}
+
+func (c *showCredentialCommand) Info() *cmd.Info {
+	return &cmd.Info{
+		Name:    "show-credential",
+		Args:    "[<cloud name> <credential name>]",
+		Purpose: "Shows credential information on a controller.",
+		Doc:     showCredentialDoc,
+		Aliases: []string{"show-credentials"},
+	}
+}
+
+func (c *showCredentialCommand) Run(ctxt *cmd.Context) error {
+	client, err := c.newAPIFunc()
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+
+	if client.BestAPIVersion() < 2 {
+		ctxt.Infof("Controller does not support credential content lookup")
+		return nil
+	}
+	contents, err := client.CredentialContents(c.CloudName, c.CredentialName, c.ShowSecrets)
+	if err != nil {
+		ctxt.Infof("Getting credential content failed with: %v", err)
+		return err
+	}
+	return c.parseContents(ctxt, contents)
+}
+
+type CredentialContentAPI interface {
+	CredentialContents(cloud, credential string, withSecrets bool) ([]params.CredentialContentResult, error)
+	BestAPIVersion() int
+	Close() error
+}
+
+func (c *showCredentialCommand) NewCredentialAPI() (CredentialContentAPI, error) {
+	currentController, err := c.store.CurrentController()
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return nil, errors.New("there is no active controller")
+		}
+		return nil, errors.Trace(err)
+	}
+	api, err := c.NewAPIRoot(c.store, currentController, "")
+	if err != nil {
+		return nil, errors.Annotate(err, "opening API connection")
+	}
+	return apicloud.NewClient(api), nil
+}
+
+type CredentialContent struct {
+	AuthType string `yaml:"auth-type" json:"auth-type"`
+	Cloud    string `yaml:"cloud" json:"cloud"`
```

Comment:
We would normally expect comments for all fields (or none).
There's a few places where comments on exported structs/interfaces are missing which would be good to have.

Reply:
Actually, most of the places in cmd package do not have comments specifically because the code is self-explanatory. Comments will repeat the information that code already conveys... Kind of defies the point of comments  :)